test/rpc_test.py

```diff
@@ -1132,15 +1116,22 @@ def test_local_shutdown_with_rpc(self):
         )
         n = self.rank + 1
         dst_rank = n % self.world_size
-        ret = rpc.rpc_sync(
+        rpc.rpc_sync(
             "worker{}".format(dst_rank),
             torch.add,
             args=(torch.ones(n, n), torch.ones(n, n)),
         )
-        # wait for RPCs to be done, so that some workers don't try to shut down
-        # too early.
-        rpc.rpc_sync("worker{}".format(dst_rank), _set_rpc_done, args=(1,))
-        _check_rpc_done(1)
+        # This is for the below `dist.barrier`.
+        # For `RpcAgent` other than `ProcessGroupAgent`,
+        # no `_default_pg` is initialized.
+        if not dist.is_initialized():
+            dist.init_process_group(
+                backend="gloo",
+                init_method=self.init_method,
+                rank=self.rank,
+                world_size=self.world_size,
+            )
+        dist.barrier()
```

Comment:
I think it's easier to understand than an ad-hoc helper function, `_check_rpc_done`, put in this file.

Still not too perfect to initialize process group. I think later, if `_wai_all_workers()` supports calling multiple times. We could use it.